### PR TITLE
Use custom error message when running out of syncblocks

### PR DESCRIPTION
**Pull request by jkotas**
Contributes to #49215

### CODE REVIEW

**Comment by jkotas** [comment]
```
System.OutOfMemoryException: Internal limitation: too many SyncBlocks.
   at System.Threading.Monitor.ObjWait(Int32 millisecondsTimeout, Object obj)
   at Program.<Main>$(String[] args)
```

File: src/coreclr/dlls/mscorrc/mscorrc.rc [hunk 1]
```diff
@@ -495,6 +495,7 @@ BEGIN
 
     IDS_EE_ARRAY_DIMENSIONS_EXCEEDED        "Array dimensions exceeded supported range."
 
+    IDS_EE_OUT_OF_SYNCBLOCKS                "Internal limitation: too many SyncBlocks."
```
Comment (mangod9) [inline src/coreclr/dlls/mscorrc/mscorrc.rc:498]: can the message mention the limit which might indicate a possible leak within the application? Appears that the limit is `2^26`?

**Review by mangod9 — APPROVED** [review]

File: src/coreclr/dlls/mscorrc/mscorrc.rc (hunk 1 above)
Comment (jkotas) [reply src/coreclr/dlls/mscorrc/mscorrc.rc:498]:
```suggestion
    IDS_EE_OUT_OF_SYNCBLOCKS                "Internal limitation: attempt to create more than 2^26 SyncBlocks."
```
How about this?

Comment (mangod9) [reply src/coreclr/dlls/mscorrc/mscorrc.rc:498]: looks good.

**Review by stephentoub — APPROVED** [review]

**Pull request merged**